### ISSUE TITLE
[Potential NAN bug] Loss may become NAN during training

### ISSUE THREAD

**Issue by Justobe** [DreamGalaxy/DeepLearningTest#1]
Hello~

Thank you very much for sharing the code, which really helps me a lot.

But I find some problem when I use the code in

`DeepLearningTest/Softmax.py`

After some iterations, it is found that the training loss becomes NAN. After carefully checking the code, I found that the following code may trigger NAN in loss:

```
exp = tf.exp(o)
```

which is in line 45.

If elements in `o` >= 88, the result of `tf.exp(o)` is inf. And this may cause the result of loss to become NAN.

It could be fixed by making the following change:

```
exp = tf.exp(tf.minimum(o, 87))
```

Hope to hear from you ~

Thanks in advance! : )

**Comment by Justobe** [comment]
@DreamGalaxy  :)
